pkg/controller/nodelifecycle/node_lifecycle_controller_test.go

```diff
@@ -167,561 +166,685 @@ func newNodeLifecycleControllerFromClient(
-		{
-			fakeNodeHandler: &testutil.FakeNodeHandler{
-				Existing: []*v1.Node{
-					{
-						ObjectMeta: metav1.ObjectMeta{
-							Name:              "node0",
-							CreationTimestamp: fakeNow,
-							Labels: map[string]string{
-								v1.LabelTopologyRegion:          "region1",
-								v1.LabelTopologyZone:            "zone1",
-								v1.LabelFailureDomainBetaRegion: "region1",
-								v1.LabelFailureDomainBetaZone:   "zone1",
-							},
-						},
+		"No Disruption: Node created recently without failure domain labels (happens only at cluster startup)": {
+			nodeList: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "node0",
+						CreationTimestamp: fakeNow,
 					},
-					{
-						ObjectMeta: metav1.ObjectMeta{
-							Name:              "node1",
-							CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
-							Labels: map[string]string{
-								v1.LabelTopologyRegion:          "region1",
-								v1.LabelTopologyZone:            "zone1",
-								v1.LabelFailureDomainBetaRegion: "region1",
-								v1.LabelFailureDomainBetaZone:   "zone1",
-							},
-						},
-						Status: v1.NodeStatus{
-							Conditions: []v1.NodeCondition{
-								{
-									Type:               v1.NodeReady,
-									Status:             v1.ConditionTrue,
-									LastHeartbeatTime:  metav1.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
-									LastTransitionTime: metav1.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
-								},
+					Status: v1.NodeStatus{
+						Conditions: []v1.NodeCondition{
+							{
+								Type:               v1.NodeReady,
+								Status:             v1.ConditionTrue,
+								LastHeartbeatTime:  fakeNow,
+								LastTransitionTime: fakeNow,
 							},
 						},
 					},
 				},
-				Clientset: fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*testutil.NewPod("pod0", "node0")}}),
 			},
-			daemonSets:          nil,
-			timeToPass:          0,
-			newNodeStatus:       v1.NodeStatus{},
-			secondNodeNewStatus: healthyNodeNewStatus,
-			expectedEvictPods:   false,
-			description:         "Node created recently, with no status.",
+			updatedNodeStatuses: []v1.NodeStatus{
+				healthyNodeNewStatus,
+			},
+			expectedInitialStates: map[string]ZoneState{
+				"": stateNormal,
+			},
+			expectedFollowingStates: map[string]ZoneState{
+				"": stateNormal,
+			},
 		},
-		// Node created recently without FailureDomain labels which is added back later, with no status (happens only at cluster startup).
-		{
-			fakeNodeHandler: &testutil.FakeNodeHandler{
-				Existing: []*v1.Node{
-					{
-						ObjectMeta: metav1.ObjectMeta{
-							Name:              "node0",
-							CreationTimestamp: fakeNow,
+		"No Disruption: Initially both zones down, one comes back": {
+			nodeList: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "node0",
+						CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
```

Comment:
do you want to unify all usage of "metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC)"-like to base on `fakeNow`?

Reply:
Yeah good point, changed this in bb5a3bcd7e8